src/ansys/openapi/common/_oidc.py

```diff
@@ -101,54 +91,58 @@ def __init__(
             if "scope" in self._authenticate_parameters
             else []
         )
-        self._oauth_session = OAuth2Session(
+
+        self._auth = OAuth2AuthorizationCodePKCE(
+            authorization_url=self._well_known_parameters["authorization_endpoint"],
+            token_url=self._well_known_parameters["token_endpoint"],
+            redirect_uri_port=32284,
+            audience=self._authenticate_parameters["apiAudience"] if "apiAudience" in self._authenticate_parameters else None,
             client_id=self._authenticate_parameters["clientid"],
-            redirect_uri=self._authenticate_parameters["redirecturi"],
             scope=scopes,
+            session=self._initial_session
         )
-        set_session_kwargs(self._oauth_session, self._api_session_configuration)
 
-        if "offline_access" in scopes:
-            self._configure_token_refresh()
+        self._auth.refresh_data.pop('audience', None)
+
+        self._authorized_session = requests.Session()
+
+        set_session_kwargs(self._authorized_session, self._api_session_configuration)
 
-        self._callback_server = OIDCCallbackHTTPServer()
         logger.info("Configuration complete.")
 
     def get_session_with_provided_token(
-        self, refresh_token: str, access_token: Optional[str] = None
-    ) -> OAuth2Session:
+        self, refresh_token: str
+    ):
         """Create a :class:`OAuth2Session` object with provided tokens.
 
-        This method configures a session to request an access token with the provided refresh token.
-        This will happen automatically when the first request is sent to the server. Alternatively,
-        an access token can be provided, and it will be used until it expires. After it expires,
-        the refresh token will be used to request a new access token.
+        This method configures a session to request an access token with the provided refresh token,
+        an access token will be requested immediately.
 
         Parameters
         ----------
         refresh_token : str
             Refresh token for the API server, typically a Base64-encoded JSON Web Token.
-        access_token : Optional[str]
-            Access token for the API server, typically a Base64-encoded JSON web token.
         """
         logger.info("Setting tokens...")
-        if access_token is not None:
-            if _log_tokens:
-                logger.debug(f"Setting access token: {access_token}")
-            self._oauth_session.token = {
-                "token_type": "bearer",
-                "access_token": access_token,
-            }
         if refresh_token is not None:
```

Comment:
This is a big `if` block, it might be easier to read if you handled the `is None` case first.